File: proyectoweb/src/main/java/co/edu/unicauca/competencias/proyectoweb/Teacher_module/Teacher_service/interfaces/ITeacherService.ts
import { Teacher } from "../../Teacher_core/entities/Teacher";

export interface ITeacherService {
  getAllTeachers(): Promise<Teacher[]>;
  getTeacherById(id: number): Promise<Teacher | null>;
  createTeacher(Teacher: Teacher): Promise<Teacher>;
  updateTeacher(id: number, Teacher: Teacher): Promise<Teacher | null>;
  deleteTeacher(id: number): Promise<boolean>;
}

File: proyectoweb/src/main/java/co/edu/unicauca/competencias/proyectoweb/Teacher_module/Teacher_service/interfaces/ITeacherService.java
package co.edu.unicauca.competencias.proyectoweb.Teacher_module.Teacher_service.interfaces;

import co.edu.unicauca.competencias.proyectoweb.Teacher_module.Teacher_infrastructure.persistence.DTO.TeacherDTO;

import java.util.List;
import java.util.Optional;

public interface ITeacherService {
    List<TeacherDTO> getAllTeachers();
    List<TeacherDTO> getAllActiveTeachers();
    Optional<TeacherDTO> getTeacherById(Integer id);
    Optional<TeacherDTO> getTeacherByName(String identification);
    Optional<TeacherDTO> createTeacher(TeacherDTO teacher);
    Optional<TeacherDTO> updateTeacher(Integer id, TeacherDTO teacher);
    int changeStatus(Integer id, String status);
}
